docs/source/articles/payment-tender-types.html.md

```diff
@@ -0,0 +1,54 @@
+---
+title: Payment Tender Types
+excerpt: TODO
+created_at: 2019/11/05
+---
+
+Payment Tender Types
+======================================================================
+
+TODO: document introduction and additional sections
+
+_Explain_ concepts in this doc.
+So that devs working through a specific tender type implementation doc (credit card, primary, advance payment) can make sense of the steps.
+Provide context.
+
+
+Gateway
+----------------------------------------------------------------------
+
+Each tender type relies on a gateway.
+The gateway is the object responsible for processing payments of that type.
+It's usually backed by a network service, but...
+It's often an instance of a class from the Active Merchant library, but...
+
+Some gateways are "bogus" gateways, which act like they're talking to a network service, but they use "canned" responses.
+Good for local development and manual testing because you can often control what response you get.
+For example, use a credit card ending in 1 to ensure a successful transaction or ending in 2 to get an exception.
+
+For most tender types, you want to initialize a different gateway for different use cases within the same application.
+For example, in production, you want to talk to the real network service, which requires initializing a gateway that knows how to do that, and initializing it with credentials.
+For local development, you may not care if you're talking to the real service, don't have credentials, and want the predictability and extra features of a bogus gateway.
+For automated testing, you might be fine using a bogus gateway, but for some or all tests you probably want to talk to the real service while developing the integration.
```

Comment:
This is more of a generic good practice than a "Workarea-specific" thing, but I think we could emphasize that there should be at least one test using VCR and hitting a real endpoint for each individual API call that you have to make, as the bogus gateway is typically not maintained as well as the "real" gateway. Furthermore, activemerchant integrations typically just use the `AM::BogusGateway` in development, which doesn't support extra methods or options needed by e.g. PayPal and others.

If you feel like this is better off in the "implementing a gateway" doc then that's cool too :-)